contrib/objcwrapper/src/objc_generator.nit

```diff
@@ -0,0 +1,233 @@
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# Code generation
+module objc_generator
+
+import opts
+
+import objc_model
+
+redef class Sys
+	# Path to the output file
+	var opt_output = new OptionString("Output file", "-o")
+end
+
+class CodeGenerator
+	# Merge the calls to `alloc` and `init...` in a single constructor?
+	#
+	# If `true`, also the default behavior, initializing an extern Objective-C object looks like:
+	# ~~~
+	# var o = new NSArray.init_with_array(some_other_array)
+	# ~~~
+	#
+	# If `false`, the object must first be allocated and then initialized.
+	# This is closer to the Objective-C behavior:
+	# ~~~
+	# var o = new NSArray
+	# o.init_with_array(some_other_array)
+	# ~~~
+	var init_with_alloc = true is writable
+
+	# Generate Nit code to wrap `classes`
+	fun generate(classes: Array[ObjcClass])
+	do
+		# Open specified path or stdin
+		var file
+		var path = opt_output.value
+		if path != null then
+			if path.file_extension != "nit" then
+				print_error "Warning: output file path does not end with '.nit'"
+			end
+
+			file = new FileWriter.open(path)
+		else
+			file = stdout
+		end
+
+		# Generate code
+		file.write "import cocoa::foundation\n\n"
+		for classe in classes do
+			write_class(classe, file)
+		end
+
+		if path != null then file.close
+	end
+
+	private fun type_convertor(type_word: String): String
+	do
+		var types = new HashMap[String, String]
+		types["char"] = "Byte"
+		types["short"] = "Int"
+		types["short int"] = "Int"
+		types["int"] = "Int"
+		types["long"] = "Int"
+		types["long int"] = "Int"
+		types["long long"] = "Int"
+		types["long long int"] = "Int"
+		types["float"] = "Float"
+		types["double"] = "Float"
+		types["long double"] = "Float"
+
+		types["NSUInteger"] = "Int"
+		types["BOOL"] = "Bool"
+		types["id"] = "NSObject"
+
+		if types.has_key(type_word) then
+			return types[type_word]
+		else
+			return type_word
+		end
+	end
+
+	private fun write_class(classe: ObjcClass, file: Writer)
+	do
+		var commented_methods = new Array[ObjcMethod]
+		file.write "extern class " + classe.name + """ in "ObjC" `{ """ + classe.name  + """ * `}\n"""
+		for super_name in classe.super_names do
+			file.write """	super """ + super_name + "\n"
+		end
+		if classe.super_names.is_empty then file.write """	super NSObject\n"""
+		write_constructor(classe, file)
+		file.write "\n"
+		for attribute in classe.attributes do
+			write_attribute(attribute, file)
+		end
+		for method in classe.methods do
+			if method.is_commented then
+				commented_methods.add(method)
+			else
+				if init_with_alloc and method.params.first.name.has("init") then continue
+				file.write """	"""
```

Comment:
Yeah... I know, but I will rewrite a lot of this in future PR. I'm thinking of merging the backend with `jwrapper`.